homeassistant/components/august/__init__.py

```diff
@@ -456,8 +463,33 @@ def _update_locks_detail(self):
 
     def lock(self, device_id):
         """Lock the device."""
-        return self._api.lock(self._access_token, device_id)
+        return _call_api_operation_that_requires_bridge(
+            self.get_lock_name(device_id),
+            "lock",
+            self._api.lock,
+            self._access_token,
+            device_id,
+        )
 
     def unlock(self, device_id):
         """Unlock the device."""
-        return self._api.unlock(self._access_token, device_id)
+        return _call_api_operation_that_requires_bridge(
+            self.get_lock_name(device_id),
+            "unlock",
+            self._api.unlock,
+            self._access_token,
+            device_id,
+        )
+
+
+def _call_api_operation_that_requires_bridge(
```

Comment:
This is a great opportunity to use a decorator.

```python
from functools import wraps

def _handle_errors(func):
    @wraps(func)
    def with_error_handling(*args, **kwargs):
        try:
            func(*args, **kwargs)
        except AugustAPIHTTPError:
            …

    return with_error_handling

class:
    @_handle_errors
    def lock(self, device_id):
```

Oh, I see now that you can't print the device name then.

Reply:
Going to leave as-is